refactor_me.py

```diff
@@ -2,25 +2,49 @@
 
 from collections import namedtuple, defaultdict
 from operator import itemgetter
+from sys import stdout
+import io
 
-Expense = namedtuple('Expense', ('type_', 'amount'))
+Expense = namedtuple('Expense', ('category', 'amount'))
 
 
 def sum_expenses(expenses, min_amount=0):
+    """Aggregates all Expense entries into a categorized list of sums
+
+    Args:
+        expenses (iter(Expense)): Iterable of Expenses to be summed
+        min_amount (int): Minimum value to match to be included with the result
+
+    Returns:
+        dict: Category-grouped results of summed values
+    """
     aggregated_expenses = defaultdict(int)
-    for expense in expenses:
-        amount = expense.amount
+    for category, amount in expenses:
         if amount >= min_amount:
-            aggregated_expenses[expense.type_] += amount
+            aggregated_expenses[category] += amount
     return aggregated_expenses
 
 
-def print_expenses(expenses):
-    for expense, amount in sorted(expenses.items(), key=itemgetter(1)):
-        print(expense, amount)
+def format_dict(input_dict, write_out=None):
+    """Takes a dictionary and prints the results to the command line or other output stream
+
+    Args:
+        input_dict (dict): Values to output to command line
+        write_out (io.IOBase): Output object that receives the formatted string, defaults
+            writing to terminal
+    """
+    key_size = max([len(key) for key in input_dict.keys()])
```

Comment:
list comprehension is excessive here, generator expression would be enough

`max(len(key) for key in input_dict)`

Reply:
Of course, this would easily reduce the memory overhead.